### PR TITLE
Aave PCV Deposit

### PR DESCRIPTION
**Pull request by Joeysantoro**
An Aave PCV deposit similar to the Compound one. Uses WethPCVDeposit to support ETH deposits into Aave

### CODE REVIEW

File: contracts/pcv/aave/AavePCVDeposit.sol [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        override
+        onlyPCVController
+    {
+        lendingPool.withdraw(address(token), amountUnderlying, to);
+        emit Withdrawal(msg.sender, to, amountUnderlying);
+    }
+
+    /// @notice returns total balance of PCV in the Deposit
+    function balance() public view override returns (uint256) {
+        return aToken.balanceOf(address(this));
```
Comment (eswak) [inline contracts/pcv/aave/AavePCVDeposit.sol:91]: is that the balance of underlying tokens, or the amount of aTokens ?

Comment (eswak) [reply contracts/pcv/aave/AavePCVDeposit.sol:91]: nvm :

image

File: contract-addresses/mainnetAddresses.json [hunk 2]
```diff
@@ -65,5 +65,10 @@
     "rariPool8FeiIrmAddress": {"address": "0x8f47be5692180079931e2f983db6996647aba0a5" },
     "rariPool8TribeIrmAddress": {"address": "0x075538650a9c69ac8019507a7dd1bd879b12c1d7" },
     "rariPool8EthIrmAddress": {"address": "0xbab47e4b692195bf064923178a90ef999a15f819" },
-    "rariPool8DaiIrmAddress": {"address": "0xede47399e2aa8f076d40dc52896331cba8bd40f7" }
+    "rariPool8DaiIrmAddress": {"address": "0xede47399e2aa8f076d40dc52896331cba8bd40f7" },
+    "aETHAddress" : {"address": "0x030bA81f1c18d280636F32af80b9AAd02Cf0854e"},
```
Comment (eswak) [inline contract-addresses/mainnetAddresses.json:69]: technically it's aWETH 🤓

File: deploy/aavePCVDeposit.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  const aaveEthPCVDeposit = await AavePCVDeposit.new(
+    coreAddress,
+    aaveLendingPool,
+    wethAddress,
+    aETHAddress,
+    aaveIncentivesController,
+    { from: deployAddress }
+  );
+  logging ? console.log('Aave ETH PCV Deposit deployed to: ', aaveEthPCVDeposit.address) : undefined;
```
Comment (eswak) [inline deploy/aavePCVDeposit.js:26]: If you want, you could also write `logging && console.log(...` & not have to bother with `: undefined` at the end

File: end-to-end/e2e.spec.ts [hunk 4]
```diff
@@ -298,6 +298,38 @@ describe('e2e', function () {
...
+      const amount = tenPow18.mul(toBN(200));
+      await web3.eth.sendTransaction({from: deployAddress, to: aaveEthPCVDeposit.address, value: amount });
+
+      const aaveBalanceBefore = await contracts.stAAVE.balanceOf(aaveEthPCVDeposit.address);
+      await aaveEthPCVDeposit.deposit();
+
+      await aaveEthPCVDeposit.claimRewards();
+      const aaveBalanceAfter = await contracts.stAAVE.balanceOf(aaveEthPCVDeposit.address);
+
+      expect(aaveBalanceAfter.sub(aaveBalanceBefore)).to.be.bignumber.greaterThan(toBN(0));
```
Comment (eswak) [inline end-to-end/e2e.spec.ts:329]: How come the rewards balance is not 0 if the deposit just happened ?

I guess you could add a small time fast-forward in your test, to check that you earn some interest?

File: contracts/pcv/aave/AavePCVDeposit.sol (hunk 1 above)
Comment (Joeysantoro) [reply contracts/pcv/aave/AavePCVDeposit.sol:91]: Good call, shouldn't expect devs to know aTokens are rebasing

File: end-to-end/e2e.spec.ts (hunk 4 above)
Comment (Joeysantoro) [reply end-to-end/e2e.spec.ts:329]: Time did fast forward at least 1 second/block between the deposit and the claim

**Pull request merged**